projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { TabPanel } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { HelpTab } from './help-tab';
+import { OverviewTab } from './overview-tab';
+import { ProductsTab } from './products-tab';
+import styles from './styles.module.scss';
+
+const tabComponentMap = {
+	overview: OverviewTab,
+	products: ProductsTab,
+	help: HelpTab,
+};
+
+/**
+ * My Jetpack Tab panel component.
+ *
+ * @return The rendered component.
+ */
+export function MyJetpackTabPanel() {
+	const params = useParams();
+	const navigate = useNavigate();
+
+	const onTabSelect = useCallback(
+		( tabName: string ) => {
+			if ( tabName !== params.tab ) {
+				navigate( `/${ tabName }` );
+			}
+		},
+		[ navigate, params.tab ]
+	);
+
+	const tabRenderer = useCallback( ( tab: { name: string; title: string } ) => {
```

Comment:
We can benefit from more explicit typing here.

Reply:
Do you have any suggestions about it?